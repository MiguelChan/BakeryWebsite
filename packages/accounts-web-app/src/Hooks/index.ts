export * from './UseGetAccounts/UseGetAccounts';
export * from './UseCreateAccount/UseCreateAccount';
