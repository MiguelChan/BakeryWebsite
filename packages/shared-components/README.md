# `@mgl/shared-components`

> TODO: description

## Usage

```
const sharedComponents = require('@mgl/shared-components');

// TODO: DEMONSTRATE API
```
