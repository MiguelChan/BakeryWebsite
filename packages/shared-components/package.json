{
  "name": "@mgl/shared-components",
  "version": "1.1.0",
  "description": "The WebApp components library",
  "main": "./build/src/index.js",
  "types": "./build/src/index.d.ts",
  "files": [
    "./build/src/index.js"
  ],
  "scripts": {
    "clean": "rm -rf build",
    "build": "tsc",
    "test": "jest --config jest.config.js",
    "lint": "eslint . --ext .ts",
    "lint:fix": "eslint . --ext .ts --fix",
    "release": "npm run clean && npm run lint && npm run test && npm run build"
  },
  "devDependencies": {
    "eslint-plugin-modules-newline": "^0.0.6"
  },
  "peerDependencies": {}
}
