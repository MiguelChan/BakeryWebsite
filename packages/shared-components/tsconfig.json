{
    "compilerOptions": {
      "target": "ESNext",
      "module": "commonjs",
      "outDir": "./build",
      "declarationDir": "build",
      "declaration": true,
      "lib": ["ESNext"],
      "types": ["node", "jest"]
    },
    "include": [
      "src/**/*",
      "tst/**/*"
    ],
    "exclude": [
        "node_modules",
        "build"
    ]
  }