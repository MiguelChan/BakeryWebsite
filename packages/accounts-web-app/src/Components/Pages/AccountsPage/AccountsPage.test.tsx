import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import { 
  AccountsPage, 
  AccountsPageProps,
} from './AccountsPage';
import {
  EmptyTable,
  LoadingTable,
  WithData,
} from './AccountsPage.stories';
import { MemoryRouter } from 'react-router-dom';
import { UseGetAccountsState } from '../../../Hooks';
import { AccountType } from '@mgl/shared-components';

describe('AccountsPage', () => {
  const setupComponent = (props: AccountsPageProps): RenderResult => {
    return render(
      <MemoryRouter>
        <AccountsPage {...props} />
      </MemoryRouter>
    );
  };

  it('Should display the loading page', () => {
    const loadingPageProps: AccountsPageProps = {
      appContext: {
        useGetAccounts: (): UseGetAccountsState => ({
          accounts: [],
          isLoading: true,
        }),
      }
    };
    
    const result = setupComponent(loadingPageProps);

    expect(result.container).toMatchSnapshot();
  });

  it('Should display the Data', () => {
    const props: AccountsPageProps = {
      appContext: {
        useGetAccounts: (): UseGetAccountsState => ({
          isLoading: false,
          accounts: [
            {
              accountType: AccountType.Entry,
              id: 'Some-id',
              subAccounts: [],
              title: 'Some TItle',
            },
          ],
        }),
      },
    };
    
    const result = setupComponent(props);

    expect(result.container).toMatchSnapshot();
  });

  it('Should display the empty table', () => {
    const props: AccountsPageProps = EmptyTable.args!;

    const result = setupComponent(props);

    expect(result.container).toMatchSnapshot();
  });
});