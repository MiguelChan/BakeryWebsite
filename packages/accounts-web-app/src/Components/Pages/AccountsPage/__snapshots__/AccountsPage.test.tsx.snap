// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsPage Should display the Data 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  />
</div>
`;

exports[`AccountsPage Should display the empty table 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  />
</div>
`;

exports[`AccountsPage Should display the loading page 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  />
</div>
`;
